# **Homelessness - Project 1**
### Analyzing qualitative and quantitative variables.

# **Importing Necessary Python Modules**

Python incorporates a variety of open source add-ins called **modules** that add extra features to the basic setup. The name of the modules is after the `import` statement, and the purpose is in a non-code comment after thew hashtag (#).



In [ ]:
import pandas as pd                 #Data analysis
import numpy as np                  #Calculations
import plotly.express as px         #Graphing
import matplotlib.pyplot as plt     #Graphing
from IPython.display import Image   #Display images
import warnings                     #Ignore version warnings
warnings.simplefilter('ignore', FutureWarning)


In [ ]:
# Replace 'image_url' with the URL of the image you want to display
image_url = 'https://endhomelessness.org/wp-content/uploads/2019/03/homelessness-statistics.jpg'

# Display the image
Image(url=image_url)

# **Context**

The Continuum of Care (CoC) Program is designed to promote communitywide commitment to the goal of ending homelessness; provide funding for efforts by nonprofit providers, and State and local governments to quickly rehouse homeless individuals and families while minimizing the trauma and dislocation caused to homeless individuals, families, and communities by homelessness; promote access to and effect utilization of mainstream programs by homeless individuals and families; and optimize self-sufficiency among individuals and families experiencing homelessness. For more information on the Program, please visit https://www.hudexchange.info/programs/coc/

The U.S. Department of Housing and Urban Development (HUD) provides Point-in-Time (PIT) count reports of sheltered and unsheltered persons experiencing homelessness, by household type and subpopulation. This data is available at the national and state level, and for each CoC. HUD also provides Housing Inventory Count (HIC) reports, which provide a snapshot of a CoC’s inventory of beds and units available on the night designated for the count by program type, and include beds dedicated to serve persons who are homeless as well as persons in Permanent Supportive Housing. This raw data set contains PIT estimates of homelessness, and the corresponding accompanying HIC data from 2021.

Attribution: Adapted from U.S. Department of Housing and Urban Development


# **About the Dataset**

This dataset contains 100 rows corresponding to a random sample of localities (typically counties or similar large regions) that received Continuum of Care funding from HUD. A total of 21 variables are provided as listed in the table below.

| Variable Name(s)         | Description |
|:-------------------------|:--- |
| CoC Number<br>CoC Name | CoC locality identifier |
| CoC Category             | Community setting designation: Rural, Urban, etc |
| Type of Count            | Description of whether fully or partially sheltered individuals (or both) are included |
| Overall Homeless, 2021   | Total homeless count in all facilities indicated by the “Type of Count” variable |
| HMIS Participation Rate for Year-Round Beds (ES,TH,SH) | Proportion of facilities in the locality that participate in the Homeless Management Information System (HMIS) |
| Total Year-Round Beds (ES, TH, SH)<br>Total Year-Round Beds (ES)<br>Total Year-Round Beds (TH)<br>Total Year-Round Beds (SH) | (all types combined)<br>ES = Emergency Shelter<br>TH = Transitional Housing<br>SH = Safe Haven |
| Total Units for Households with Children (ES, TH, SH)<br>Total Beds for Households with Children (ES, TH, SH) | Number of units and beds designated for households with children |
| Sheltered ES Homeless 2021 | Estimate of the number of individuals sheltered in an Emergency Facility at the time of the study |
| Sheltered ES Homeless - Age 18 to 24, 2021 | Estimate of the number of individuals sheltered in an Emergency Facility by age |
| Sheltered ES Homeless - Female, 2021<br>Sheltered ES Homeless - Male, 2021<br>Sheltered ES Homeless - Trans+, 2021 | Estimate of the number of individuals sheltered in an Emergency Facility by gender |
| Sheltered ES Homeless - Hispanic/Latino, 2021<br>Sheltered ES Homeless - White, 2021<br>Sheltered ES Homeless - Black or African American, 2021<br>Sheltered ES Homeless - Asian or Pacific Islander<br>Sheltered ES Homeless - American Indian or Alaska Native, 2021<br>Sheltered ES Homeless - Multiple Races, 2021 | Estimate of the number of individuals sheltered in an Emergency Facility by race/ethnicity |



Let's take a look at the data. To do this, first we import it directly from the url below.



# **A Snippet of the Data**

In [ ]:
url='https://raw.githubusercontent.com/thamilton562/STAT108_Projects_Students/main/DataSets/Homelessness.csv'
df=pd.read_csv(url)

Next, we can display the data by *typing the name* of the DataFrame. To ensure we can see all columns, we'll use the *pd.set_option* method.

In [ ]:
# Set display options to show all columns
pd.set_option('display.max_columns', None)
df

CoC Number                                           CoC Name  \
0      LA-505                     Monroe/Northeast Louisiana CoC   
1      MN-502                  Rochester/Southeast Minnesota CoC   
2      PA-508                     Scranton/Lackawanna County CoC   
3      OK-503                      Oklahoma Balance of State CoC   
4      OH-504                     Youngstown/Mahoning County CoC   
..        ...                                                ...   
95     KS-507                        Kansas Balance of State CoC   
96     CA-507                                   Marin County CoC   
97     TN-512  Morristown/Blount, Sevier, Campbell, Cocke Cou...   
98     MA-515                                     Fall River CoC   
99     CA-526  Amador, Calaveras, Mariposa, Tuolumne Counties...   

               CoC Category                   Type of Count  \
0         Largely Rural CoC            Sheltered-Only Count   
1         Largely Rural CoC            Sheltered-Only Count   
2      Largely Suburban CoC  Sheltered and full unsheltered   
3         Largely Rural CoC            Sheltered-Only Count   
4   Other Largely Urban CoC            Sheltered-Only Count   
..                      ...                             ...   
95        Largely Rural CoC            Sheltered-Only Count   
96     Largely Suburban CoC            Sheltered-Only Count   
97        Largely Rural CoC  Sheltered and full unsheltered   
98     Largely Suburban CoC  Sheltered and full unsheltered   
99        Largely Rural CoC            Sheltered-Only Count   

    Overall Homeless, 2021  \
0                       78   
1                      419   
2                      165   
3                      125   
4                       62   
..                     ...   
95                     783   
96                     225   
97                     472   
98                     324   
99                     151   

    HMIS Participation Rate for Year-Round Beds (ES, TH, SH)  \
0                                               0.233          
1                                               0.604          
2                                               0.839          
3                                               0.295          
4                                               0.192          
..                                                ...          
95                                              0.162          
96                                              0.727          
97                                              0.316          
98                                              0.877          
99                                              0.598          

    Total Year-Round Beds (ES, TH, SH)  Total Year-Round Beds (ES)  \
0                                  133                          69   
1                                  599                         378   
2                                  155                          64   
3                                  149                         135   
4                                  198                         187   
..                                 ...                         ...   
95                                1170                         703   
96                                 293                         163   
97                                 133                         117   
98                                 236                         212   
99                                 214                         156   

    Total Year-Round Beds (TH)  Total Year-Round Beds (SH)  \
0                           64                           0   
1                          221                           0   
2                           80                          11   
3                           14                           0   
4                           11                           0   
..                         ...                         ...   
95                      

# **INSTRUCTIONS**

* Use Python to analyze the data set and complete each of the following.
* Replace ellipsis (...) with the relavent names or code.  
* For problems that require a written response, double click the text box to start typing.
* Reference the 3 tutorials from activity for assistance.
* Attend office hours if you still need help.

## **QUESTION 1**
Determine whether the four variables below are qualitative or quantitative. If they are quantitative, specify whether they are continuous or discrete.

| Variable                   | Classification            |
|:---------------------------|:--------------------------|
| CoC Category               | Qualitative               |
| Type of Count              | Qualitative               |
| HMIS Participation Rate    | Quantitative, continuous  |
| Sheltered ES Homeless 2021 | Quantitative, discrete    |

## **QUESTION 2**

For question 2 you will analyze a qualitative variable. Find your variable based on your last name and use that variable when answering all parts of question 2.

Once you find your variable description, scroll up to "About the Dataset" to find the variable name. Then look at the "Snippet of Data" to get the exact variable name, especially since variable names are case sensitive. You can scroll to the right, if needed.

Construct a frequency table, relative frequency table, and relative frequency bar chart to describe the distribution of your variable. State any fact that jumps out to you.

| **Last Name** | **Variable Description** |
|:--------------|:-------------------------|
| A-L           | CoC Category             |
| M-Z           | Type of Count            |



**2a)** Construct a table that contains the frequency and relative frequency distribution for your variable. Round relative frequency to 3 decimal places.

# **A-L**

In [ ]:
# Define the name of the variable to be analyzed
variable = df['CoC Category']               #variable = df['..']

# Create the frequency table and sort the categories in numerical order.
# .sort_index works here because the category names are numerical.
# rename "count" to "frequency"
freq_table = pd.value_counts(variable)
freq_table = freq_table.rename('Frequency')

# Create the relative frequency table, and rename the counts column to
#   Relative Frequency.
relative_freq_table = freq_table/len(df)                                            #relative_freq_table=freq_table/... HINT: look back at Project 0 or Tutorial 1.
relative_freq_table = relative_freq_table.rename('Relative Frequency').round(3)     # relative_freq_table = relative_freq_table.rename('...').round(...)

# Combine both tables
# axis=1 says to put the tables together as columns
combined_table=pd.concat([freq_table, relative_freq_table], axis=1)                 # combined_table=pd.concat([..., ...], axis=1)

# Print the combined table.
combined_table                                                          # ...


Frequency  Relative Frequency
CoC Category                                          
Largely Suburban CoC            43                0.43
Largely Rural CoC               33                0.33
Other Largely Urban CoC         13                0.13
Major City CoC                  11                0.11

# **M-Z**

In [ ]:
# Define the name of the variable to be analyzed
variable = df['Type of Count']               #variable = df['..']

# Create the frequency table and sort the categories in numerical order.
# .sort_index works here because the category names are numerical.
# rename "count" to "frequency"
freq_table2 = pd.value_counts(variable)
freq_table2 = freq_table2.rename('Frequency')

# Create the relative frequency table, and rename the counts column to
#   Relative Frequency.
relative_freq_table2 = freq_table2/len(df)                                            #relative_freq_table=freq_table/... HINT: look back at Project 0 or Tutorial 1.
relative_freq_table2 = relative_freq_table2.rename('Relative Frequency').round(3)     # relative_freq_table = relative_freq_table.rename('...').round(...)

# Combine both tables
combined_table2=pd.concat([freq_table2, relative_freq_table2], axis=1)                 # combined_table=pd.concat([..., ...], axis=1)

# Print the combined table.
combined_table2                                                          # ...


Frequency  Relative Frequency
Type of Count                                                   
Sheltered-Only Count                      43                0.43
Sheltered and full unsheltered            33                0.33
Sheltered and partial unsheltered         24                0.24

**2b)** Construct a relative frequency bar chart to describe the distribution of chest pain type.

# **A-L**

In [ ]:
# The argument in (...) tells the system which table to use to make the
#      bar graph. Since we want a relative freqency bar chart, which is the
#      correct label to replace the ellipsis (...) in this code?
dfrf = pd.DataFrame(relative_freq_table)  #dfrf = pd.DataFrame(...)

# Create the bar graph
fig = px.bar(x=dfrf.index,y=dfrf['Relative Frequency'],
             title='Relative Frequency Distribution Bar Chart') #title=...,

# Update axis labels
fig.update_layout(xaxis_title="CoC Category")        #xaxis_title=...
fig.update_layout(yaxis_title="Relative Frequency")             #yaxis_title=...

# Show the bar graph
fig.show()

# **M-Z**

In [ ]:
# The argument in (...) tells the system which table to use to make the
#      bar graph. Since we want a relative freqency bar chart, which is the
#      correct label to replace the ellipsis (...) in this code?
dfrf2 = pd.DataFrame(relative_freq_table2)  #dfrf = pd.DataFrame(...)

# Create the bar graph
fig2 = px.bar(x=dfrf2.index,y=dfrf2['Relative Frequency'],
             title='Relative Frequency Distribution Bar Chart') #title=...,

# Update axis labels
fig2.update_layout(xaxis_title="Type of Count")        #xaxis_title=...
fig2.update_layout(yaxis_title="Relative Frequency")             #yaxis_title=...

# Show the bar graph
fig2.show()

**2c)** Describe the distribution of your variable.

**A-L:** Localities are most likely to be largely suburban, followed by largely rural, other largely urban, and major city.

**M-Z:** Count types are most likely to be shelter-only, followed by sheltered and full unsheltered, and sheltered and partial unsheltered.

## **Question 3**

For question 3 you will analyze a quantitative variable. Find your variable based on your last name and use that variable when answering all parts of question 3.  

Once you find your variable description, scroll up to "About the Dataset" to find the variable name. Then look at the "Snippet of Data" to get the exact variable name, especially since variable names are case sensitive.

| **Last Name** | **Variable Description**               |
|---------------|----------------------------------------|
| A-L           | HMIS Participation Rate for Year-Round Beds (ES, TH, SH) |
| M-Z           | Sheltered ES Homeless, 2021   |



**3a)** Construct a histogram for your variable. Use number of bins = 18.

# **A-L**

In [ ]:
# Create the histogram, with the x-axis being the variable specified in the
#   table based on your last name.
fig = px.histogram(x=df['HMIS Participation Rate for Year-Round Beds (ES, TH, SH)'], nbins=18,                         #fig = px.histogram(x=df['...'],nbins = ...,
             title='Histogram of HMIS Participation Rates',                                                            #title='...',
             labels={'x':'Proportion of Facilities That Participate In the Homeless Management Information System'})  #labels={'x':'...')

# Update the vertical axis title.
fig.update_layout(yaxis_title='Frequency')                    #fig.update_layout(yaxis_title='...')

# Print the histogram
fig.show()

# **M-Z**

In [ ]:
# Create the histogram, with the x-axis being the variable specified in the
#   table based on your last name.
fig = px.histogram(x=df['Sheltered ES Homeless, 2021'],nbins = 18,                           #fig = px.histogram(x=df['...'],nbins = ...,
             title='Histogram of Individuals Sheltered In An Emergency Facility',                  #title='...',
             labels={'x':'Individuals Sheltered In An Emergency Facility'})                        #labels={'x':'...')

# Update the vertical axis title.
fig.update_layout(yaxis_title='Frequency')                       #fig.update_layout(yaxis_title="...")

# Print the histogram
fig.show()

**3b)** Construct a boxplot for your variable.  

# **A-L**

In [ ]:
# Create the boxplot, with a title, and specify horizontal axis label.
px.box(x=df['HMIS Participation Rate for Year-Round Beds (ES, TH, SH)'],             #px.box(x=df[...],
       title='Boxplot of HMIS Participations Rates',                 #title='...',
       labels={'x':'Proportion of Facilities That Participate In the Homeless Management Information System'})                     #labels={'x':'...'})

# **M-Z**

In [ ]:
px.box(x=df['Sheltered ES Homeless, 2021'],
       title='Boxplot of Individuals Sheltered In An Emergency Facility',
       labels={'x':'Individuals Sheltered In An Emergency Facility'})

**3c)** Calculate the following summary statistics for your variable: 5 number summary, mean, and standard deviation. Round to three decimal places.

In [ ]:
# Calculate the numerical summaries
# Indicate your variable.
descriptive_stats = df[['HMIS Participation Rate for Year-Round Beds (ES, TH, SH)',
                        'Sheltered ES Homeless, 2021']].describe().round(3)       #descriptive_stats = df[['...']].describe().round(...)

# Print the results.
descriptive_stats                                                                               #...


HMIS Participation Rate for Year-Round Beds (ES, TH, SH)  \
count                                            100.000          
mean                                               0.687          
std                                                0.224          
min                                                0.000          
25%                                                0.589          
50%                                                0.759          
75%                                                0.843          
max                                                1.000          

       Sheltered ES Homeless, 2021  
count                       100.00  
mean                        604.23  
std                        1532.56  
min                           2.00  
25%                         105.75  
50%                         199.50  
75%                         506.00  
max                       14144.00

**3d)** Use information from (3a), (3b) and 3(c) to describe your variable in terms of shape, center, spread, and outliers.
* Use the correct center and the correct spread based on the shape of the distribution.
* Specify which center and which spread you are using. For ex: Say "The mean is ..." or "The median is ...", rather than "The center is ..."
* When addressing outliers, if any, list the values of **all** outliers.
* Include units, if any, for all numbers.

**NOTE:** Students must use the center/spread that match their chosen shape. Skewed/outliers = median/IQR, Symm/no outliers = mean/sd

**A-L:** The distribution of HMIS Participation Rates is skewed left. The outliers are 0, 0.162, and 0.166. The median is 0.759. The IQR is 0.254.

**M-Z:** The distribution of the number of homeless sheltered in emergency facilities (Sheltered ES Homeless, 2021) is skewed right. There 11 outliers; 1217, 1318, 1449, 1479, 1805, 2398, 2611, 2838, 2876, 4072, 14,144 (14.144k) people. The median is 199.5 people (homeless). The IQR is 400.25 people (homeless).

**3e)** Interpret the standard deviation in context.

**A-L:** The typical HMIS participation rate falls within 0.224 of the mean HMIS participation rate.

**M-Z:** The typical number of homeless sheltered in emergency facilities (Sheltered ES Homeless, 2021) falls within 1532.56 people of the mean.

**3f)** Interpret the IQR in context.

**A-L:** The range of the middle half (50%) of HMIS participation rates is 0.254.

**M-Z:** The range of the middle half (50%) of Sheltered ES homeless, 2021, is 400.25 people.

## **QUESTION 4**

How do major city CoC's and largely rural CoC's compare with respect to whites and blacks/African Americans in emergency shelters?

Calculate the mean number of sheltered ES whites and sheltered ES blacks for major city CoC's and largely rural CoC's. Round to 2 decimal places. Compre the results. Then answer a question about the code.

**4a)** Calculate and state the number of sheltered ES whites and sheltered ES blacks for major city CoC's and largely rural CoC's.  Round to two decimal places.

In [ ]:
# major_city_means = df[df['CoC Category'] == '...'][['...', '...']].mean().round(...)
# largely_rural_means = df[df['...'] == 'Largely Rural CoC'][['...', '...']].mean().round(...)
major_city_means = df[df['CoC Category'] == 'Major City CoC'][['Sheltered ES Homeless - White, 2021',
                                                               'Sheltered ES Homeless - Black or African American, 2021']].mean().round(2)
largely_rural_means = df[df['CoC Category'] == 'Largely Rural CoC'][['Sheltered ES Homeless - White, 2021',
                                                                     'Sheltered ES Homeless - Black or African American, 2021']].mean().round(2)

# Combine the two table and specify labels.
combined_means = pd.DataFrame({'White': [major_city_means['Sheltered ES Homeless - White, 2021'], largely_rural_means['Sheltered ES Homeless - White, 2021']],
                               'Black': [major_city_means['Sheltered ES Homeless - Black or African American, 2021'], largely_rural_means['Sheltered ES Homeless - Black or African American, 2021']]},
                        index=['Major City Coc', 'Largely Rural CoC'])

# Print the table
combined_means # ...

White    Black
Major City Coc     1385.45  1251.00
Largely Rural CoC   175.55    36.45

**4b)** Compare the results for people with and without heart disease.

Major city CoC's have a larger mean number of white people and a larger mean number of black people in emergency shelters.

**4c)** In the table below are some snippets of code.

| **Last Name Initial** | **Code**                    |
|:-----------------------|:----------------------------------|
| A-L                    | df['CoC Category'] == 'Major City CoC'  |
| M-Z                    | df[df['CoC Category'] == 'Major City CoC']|


Based on your last name, interpret the snippet of code.

**A-L:** Checks to see if the CoC Category is Major City CoC.

**M-Z:** Creates a DataFrame with just Major City Coc's.


## **QUESTION 5**

Generate a paragraph of at least 100 words to address one of the following questions. That is, answer only 5a or 5b, but not both.

**5a)** Discuss how analyzing your chosen data set using statistical methods could help you become better prepared for future courses in your major?

...

--OR--

**5b)** Discuss how analyzing your chosen data set using statistical methods could be instrumental in becoming better prepared for your future career?

...


<br><br>
### Once you are done and ready to submit, follow the instructions below to save as a PDF and submit to GradeScope.

### Save as PDF
1. Run all code one last time
2. File-> Print Preview opens in a new browser window
3. Verify you can see all graphs. If not, go back to step 1.
4. File -> Print (or ctrl-p/cmnd-p)
5. Change destination to PDF (don't save, yet)
6. Scroll through preview to make sure you can see your graphs entirely. If not, click Cancel. Make the browser window narrower. Go back to step 4.
7. Repeat steps 4-6 until you can see your graphs completely. But do not make them too narrow.
8. Save the PDF, taking note of where it is saved.

### Submit to GradeScope
1. Login to the Canvas course
2. Click on GradeScope in the course navigation.
3. If you see multiple courses in GradeScope, click on the STAT 108 course
4. Click on the "Tutorial 2 Practice Upload" assignment
5. Click on "Submit Work", select PDF
6. Select the PDF you just created
7. You need to tell GradeScope which page each problem answer/output is on. You should see a list of problems on the right, and a display of pages (thumbnails) on the right.
Assign pages to questions by clicking on the question number on the left, then clicking on all pages that question is on.
8. After ALL questions have been assigned to their respective page(s), click "Submit"
